azurelinuxagent/common/cgroup.py

```diff
@@ -13,104 +13,108 @@
 # limitations under the License.
 #
 # Requires Python 2.6+ and Openssl 1.0+
-
+import os
 import re
 
+from azurelinuxagent.common import logger
 from azurelinuxagent.common.exception import CGroupsException
+from azurelinuxagent.common.future import ustr
 from azurelinuxagent.common.osutil import get_osutil
+from azurelinuxagent.common.utils import fileutil
 
-re_user_system_times = re.compile('user (\d+)\nsystem (\d+)\n')
-
-# The metric classes (Cpu, Memory, etc) can all assume that CGroups is enabled, as the CGroupTelemetry
-# class is very careful not to call them if CGroups isn't enabled. Any tests should be disabled if the osutil
-# is_cgroups_support() method returns false.
+re_user_system_times = re.compile(r'user (\d+)\nsystem (\d+)\n')
 
 
 class CGroup(object):
-    def __init__(self):
-        pass
+    def __init__(self, name, cgroup_path, controller_type):
+        self.name = name
```

Comment:
why do cgroups need a name? the path is already unique

Reply:
The Name helps me to create a new cgroup_metric of the same name, which is used in the json metric as well.

Example: in `azurelinuxagent/common/cgroupstelemetry.py:96`
`CGroupsTelemetry._cgroup_metrics[cgroup.name] = CgroupMetrics()`

Reply:
still, you could use the path as key instead of the name... the path is guaranteed to be unique... as we add new cgroups to track services, cron jobs and what not we will need to come up with a way to name these and to ensure the name remains unique... path does not have that issue.